### PR TITLE
M2P-264 Adds discount code to description in order

### PR DESCRIPTION
**Pull request by njfaries**
# Description
Minor change to M2 side, include coupon code in description to unify discount handling on hail side. Right now there is a special case just for M2 that will be able to be removed by this change. Will link related PR once created.

Fixes: M2P-264

#changelog M2P-264 Adds discount code to description in order

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: Helper/Cart.php [hunk 1]
```diff
@@ -1962,7 +1962,7 @@ public function collectDiscounts(
                         case RuleInterface::COUPON_TYPE_SPECIFIC_COUPON:
                         case RuleInterface::COUPON_TYPE_AUTO:
                             $couponCode = $quote->getCouponCode();
-                            $description = trim(__('Discount ') . $rule->getDescription());
+                            $description = trim(__('Discount ') . $rule->getDescription(). " ({$quote->getCouponCode()})");
```
Comment (danac-gs) [inline Helper/Cart.php:1965]:
```suggestion
                            $description = trim(__('Discount ') . $rule->getDescription(). " ($couponCode)");
```

**Review by danac-gs — CHANGES_REQUESTED** [review]

File: Helper/Cart.php (hunk 1 above)
Comment (njfaries) [reply Helper/Cart.php]: oops good call, missed this when merging conflict.

**Review by danac-gs — DISMISSED** [review]

**Review by vitaliyreznikov — CHANGES_REQUESTED** [review]
1. I've restored task description in jira, could you please check it?
2. You fixed the case when discount is added on magento side, could you fix discount added via bolt modal as well? This is last line of file Model/discountCodeValidation.php

File: Helper/Cart.php [hunk 2]
```diff
@@ -1962,7 +1962,7 @@ public function collectDiscounts(
                         case RuleInterface::COUPON_TYPE_SPECIFIC_COUPON:
                         case RuleInterface::COUPON_TYPE_AUTO:
                             $couponCode = $quote->getCouponCode();
-                            $description = trim(__('Discount ') . $rule->getDescription());
+                            $description = trim(__('Discount ') . $rule->getDescription(). " ({$couponCode})");
```
Comment (adentong) [inline Helper/Cart.php:1965]:
```suggestion
                            $description = trim(__('Discount ') . $rule->getDescription() . " ({$couponCode})");
```

File: Test/Unit/Helper/CartTest.php [hunk 3]
```diff
@@ -3739,7 +3739,7 @@ public function collectDiscounts_withCouponCode_collectsCouponCodeDiscount()
         $expectedTotalAmount = $totalAmount - $expectedDiscountAmount - $expectedDiscountAmountNoCoupon;
         $expectedDiscount = [
             [
-                'description' => trim(__('Discount ') . self::COUPON_DESCRIPTION),
+                'description' => trim(__('Discount ') . self::COUPON_DESCRIPTION . ' (' . self::COUPON_CODE . ')'),
```
Comment (adentong) [inline Test/Unit/Helper/CartTest.php:3742]: seems like using curly braces might be a bit more succinct? same in the two places below

Comment (adentong) [reply Test/Unit/Helper/CartTest.php:3742]: though you have to double quotes

Comment (njfaries) [reply Test/Unit/Helper/CartTest.php:3742]: maybe I just did it wrong, but that didn't like the `self` variables.

Comment (adentong) [reply Test/Unit/Helper/CartTest.php:3742]: well i'm certainly not a php expert so if it doesn't work it doesn't work.

**Review by vitaliyreznikov — APPROVED** [review]

**Review by danac-gs — APPROVED** [review]

**Pull request merged**